src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/ViewContext.cs

```diff
@@ -58,16 +58,6 @@ public ViewContext()
                 throw new ArgumentNullException(nameof(view));
             }
 
-            if (viewData == null)
```

Comment:
Expect lots of tag and HTML helpers to blow up if `ViewContext.ViewData` is `null`. May also experience NREs if `TempData` is allowed to be null here.

Why not instead leave `ViewContext` unchanged and have `ViewExecutor` new up instances if it's passed `null`?